cylc/flow/task_pool.py

```diff
@@ -362,20 +362,35 @@ def compute_runahead(self, force=False) -> bool:
             # Find the earliest point with unfinished tasks.
             for point, itasks in sorted(self.get_tasks_by_point().items()):
                 if (
-                    points  # got the limit already so this point too
-                    or any(
-                        not itask.state(
-                            TASK_STATUS_FAILED,
-                            TASK_STATUS_SUCCEEDED,
-                            TASK_STATUS_EXPIRED
-                        )
```

Comment:
This counted waiting runahead tasks.

It should be superfluous if `not cylc.flow.flags.cylc7_back_compat` because `is_incomplete` should cover it anyway.